clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -6158,12 +6158,24 @@ def err_illegal_initializer_type : Error<"illegal initializer type %0">;
 def ext_init_list_type_narrowing : ExtWarn<
   "type %0 cannot be narrowed to %1 in initializer list">,
   InGroup<CXX11Narrowing>, DefaultError, SFINAEFailure;
+// *_narrowing_const_reference diagnostics have the same messages, but are
+// controlled by -Wc++11-narrowing-const-reference for narrowing involving a
+// const reference.
+def ext_init_list_type_narrowing_const_reference : ExtWarn<
```

Comment:
There should be a test for each of these new diagnostics.

Reply:
The 3 `def` are all covered by `SemaCXX/GH63151.cpp` using the additional `-Wno-c++11-narrowing-const-reference ` RUN line